### PR TITLE
fix: fix some bad dereferencing and looping.

### PR DESCRIPTION
**Pull request by justbert**
Also update to go 1.18.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 	istioinformers "istio.io/client-go/pkg/informers/externalversions"
 	kubeinformers "k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
```
Comment (justbert) [inline main.go:13]: Needed for us to be able to run the code using the Azure KUBECONFIGs. I wonder if this is needed in the AKS-Engine build?

**Review by zachomedia — APPROVED** [review]
LGTM!

File: main.go (hunk 1 above)
Comment (zachomedia) [reply main.go:13]: I want to say it's already there since it knows to prompt for the authentication code.

Comment (justbert) [reply main.go:13]: Hmm... that's fair.

**Pull request merged**